packages/core-ethereum/crates/core-ethereum-actions/src/channels.rs

```diff
@@ -124,8 +132,17 @@ where
                     }
                 }
                 ChannelStatus::Open => {
-                    // TODO: emit "channel state change" event
+                    if redeem_before_close {
+                        // TODO: trigger aggregation
+                        // Do not await the redemption, just submit it to the queue
+                        let redeemed = redeem_tickets_in_channel(db.clone(), &channel, false, tx_sender.clone())
+                            .await?
+                            .len();
+                        info!("{redeemed} tickets will be redeemed before closing {channel}");
+                    }
+
                     tx_sender.send(Transaction::CloseChannel(channel)).await
+                    // TODO: emit "channel state change" event
```

Comment:
TODO

Reply:
That event is currently emitted by the indexer when it indexs the channel close event, so no need to address now.